tests/status.py

```diff
@@ -72,6 +72,62 @@ def runTest(self):
         board.expect(prompt)
         self.result_message = 'ifconfig shows ip addresses: %s' % tmp
 
+class MacAddressCheck(rootfs_boot.RootFSBootTest):
+    '''Checked MAC addresses for all interfaces.'''
+    def __checkMac(self, num1, num2, count):
+        if num1:
+            num1 = re.sub('[:]', '', num1.upper())[-6:]
+            result = int(num2, base=16) + count
+            if result == int(num1, base=16):
+                self.result_message = 'MAC addresses are sequential'
+            else:
+                raise ValueError('MAC addresses are not sequential')
+        else:
+            raise ValueError('Could not find appropriate MAC address in %s' % num1)
+
+    def runTest(self):
+#       TODO: enable wi-fi mac address check from /sys/class once wi-fi tests are added
+#       board.sendline('\ncat /sys/class/net/wlan0/address')
+#       board.expect('wlan0/address')
+#       board.expect(prompt)
+#       mac0 = re.search(r'00:19:f5:([a-z0-9:]{8})', board.before).group()
+#       mac0 = re.sub('[:]', '', mac0.upper())[-6:]
+#       board.sendline('\ncat /sys/class/net/wlan0-1/address')
+#       board.expect('wlan0-1/address')
+#       board.expect(prompt)
+#       mac1 = re.search(r'00:19:f5:([a-z0-9:]{8})', board.before).group()
+#       self.__checkMac(mac1, mac0, 1)
+        board.sendline('\nhexdump -n 6 -v -e \'8/1 "%02X" "\n"\' /proc/device-tree/uccp@18480000/mac-address0')
+        board.expect('mac-address0')
+        board.expect(prompt)
+        mac0 = re.search(r'0019F5([A-Z0-9]{6})', board.before).group()
+        mac0 = mac0.upper()[-6:]
+        board.sendline('\nhexdump -n 6 -v -e \'8/1 "%02X" "\n"\' /proc/device-tree/uccp@18480000/mac-address1')
+        board.expect('mac-address1')
+        board.expect(prompt)
+        mac1 = re.search(r'0019F5([A-Z0-9]{6})', board.before).group()
+        self.__checkMac(mac1, mac0, 1)
+        board.sendline('\nhcitool dev')
+        board.expect('hci0')
+        board.expect(prompt)
+        mac2 = re.search(r'00:19:F5:([A-Z0-9:]{8})', board.before).group()
```

Comment:
A minor issue, but this will look for a pattern of 8 characters that include A-Z, 0-9 or :, you could do this instead as 3 groups of (:[A-Z0-9]{2}) or something along those lines.